src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -1380,7 +1380,7 @@ internal bool TryReadInt16(out short value)
                 // If the int16 isn't fully in the buffer, or if it isn't fully in the packet,
                 // then use ReadByteArray since the logic is there to take care of that.
 
-                if (!TryReadByteArray(_bTmp, 0, 2))
+                if (!TryReadByteArray(_bTmp.AsSpan(), 2))
```

Comment:
Can you rely on the implicit cast from byte[] to Span here rather than calling AsSpan() explicitly?